pkg/resource/deploy/step_executor_test.go

```diff
@@ -103,6 +103,7 @@ func TestStepExecutor(t *testing.T) {
 
 			se := &stepExecutor{
 				deployment: &Deployment{
+					opts: &Options{},
```

Comment:
There are a few tests where we need to instantiate empty objects to avoid `nil` dereferences/panics. Is this OK? I wonder if we should be using `NewDeployment`-esque constructor functions in our tests rather than doing this, since it's harder to know whether something is wrong or whether the test is violating invariants that won't crop up in "real" code.

Reply:
This is fine, we do it a lot